### PR TITLE
Issue fix for the menu has no menu items #1223

### PR DESCRIPTION
**Pull request by JeevantheDev**
**Overview**
- Set a useState hook for looking if there are empty menu items or not.
- If there are not any items then it will render  "**No menu items...**" inside the Popover.

**Screenshot**
menu-dropdown

**Documentation**
- [x] Updated Typescript types and/or component PropTypes
- [x] Added / modified component docs
- [x] Added / modified Storybook stories

### CODE REVIEW

File: src/menu/src/Menu.js [hunk 1]
```diff
@@ -101,9 +104,19 @@ const Menu = memo(function Menu(props) {
   }, [menuRef])
```
Comment (mshwery) [inline src/menu/src/Menu.js:104]: Given that menuRef wont actually trigger this hook to run again when it changed (mutable refs wont trigger renders or hooks to run again), I would opt for us calculating `isEmpty` separately, on every render. We don't need useState for it.

**Review by mshwery — COMMENTED** [review]
looks good but curious if we could do this without useState or adding the logic to that huge useEffect hook?

File: src/menu/src/Menu.js [hunk 2]
```diff
@@ -26,7 +28,8 @@ const Menu = memo(function Menu(props) {
       : []
 
     if (menuItems.current.length === 0) {
-      throw new Error('The menu has no menu items')
+      setIsEmpty(true)
+      // throw new Error('The menu has no menu items')
```
Comment (mshwery) [inline src/menu/src/Menu.js:32]: Can we remove this commented out code?

Comment (JeevantheDev) [reply src/menu/src/Menu.js:32]: Ya ya sure.

File: src/menu/src/Menu.js (hunk 1 above)
Comment (JeevantheDev) [reply src/menu/src/Menu.js:104]: Ok, I got it. Once I do it I will do the PR.

File: src/menu/src/Menu.js (hunk 2 above)
Comment (JeevantheDev) [reply src/menu/src/Menu.js]: I have committed the code without useState hook. Kindly review it.

**Review by brandongregoryscott — APPROVED** [review]

**Pull request merged**